src/Forms/FormMessage.php

```diff
@@ -92,6 +91,57 @@ public function setMessage(
         return $this;
     }
 
+    /**
+     * Appends a message to the existing message if the types match, overrides if not
+     *
+     * @param string $message Message string
+     * @param string $messageType Message type
+     * @param string $messageCast
+     * @param bool $force Whether or not to override the existing message if appending is invalid
+     * @throws InvalidArgumentException
+     * @return $this
+     */
+    public function appendMessage(
+        string $message,
+        string $messageType = ValidationResult::TYPE_ERROR,
+        string $messageCast = ValidationResult::CAST_TEXT,
+        bool $force = false,
+    ): static {
+        if (empty($this->message)) {
```

Comment:
We should also have an early check for if `empty($message)` - and if so, just return. It protects us from accidentally overriding an existing message with an _empty_ message if the message type and cast differ from the existing one.